### PR TITLE
Scale config changes for scale prow job on AWS

### PR DESCRIPTION
**Pull request by prateekgogia**

### CODE REVIEW

**Comment by hakman** [comment]
@prateekgogia Seems that eu-central-1 is better for running the test:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/15599/presubmit-kops-aws-scale-amazonvpc-using-cl2/1678576342553071616

**Comment by hakman** [comment]
@prateekgogia Please see my previous comment. It would reduce some of the current flakes.

**Comment by prateekgogia** [comment]
@hakman Thanks yeah will update the region, I wanted to run this to make sure nothing was broken in terms of config. Ran into this error `The pod could not start because it could not mount the volume "aws-cred": secret "aws-credentials-boskos-scale-001-kops" not found`

**Comment by ameukam** [comment]
Currently `presubmit-kops-aws-scale-amazonvpc-using-cl2` running inside the cluster `default` instead of cluster `eks-prow-build-cluster` where the secrets are located.

**Comment by hakman** [comment]
Not sure why getting the SSM parameter fails, but https://github.com/kubernetes/kops/pull/15775 should help see the actual error.

**Comment by hakman** [comment]
```
Error: control-plane-us-west-2c-1.spec.image: Invalid value: "ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id":
specified image "ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id" is invalid:
failed to get value for SSM parameter: AccessDeniedException: User: arn:aws:iam::226543828060:user/kops
is not authorized to perform: ssm:GetParameter on resource: arn:aws:ssm:us-west-2::parameter/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id because
no identity-based policy allows the ssm:GetParameter action
	status code: 400, request id: 277cf3c0-f08a-4361-b4ba-12da0ffac732
```

**Comment by hakman** [comment]
@pkprzekwas @ameukam @dims I guess this is something that needs fixing on the permissions side:
```
Error: control-plane-us-west-2c-1.spec.image: Invalid value: "ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id":
specified image "ssm:/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id" is invalid:
failed to get value for SSM parameter: AccessDeniedException: User: arn:aws:iam::226543828060:user/kops
is not authorized to perform: ssm:GetParameter on resource: arn:aws:ssm:us-west-2::parameter/aws/service/canonical/ubuntu/server/20.04/stable/current/arm64/hvm/ebs-gp2/ami-id because
no identity-based policy allows the ssm:GetParameter action
	status code: 400, request id: 277cf3c0-f08a-4361-b4ba-12da0ffac732
```

**Comment by dims** [comment]
agree @hakman !

**Comment by hakman** [comment]
@dims I also think someone needs to ask for ec2 quota. At the moment this account can only create 5 instances per region (I guess the default value).

**Comment by hakman** [comment]
@prateekgogia I am proposing some changes in https://github.com/kubernetes/kops/pull/15783 that should get tests working again (more or [less](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kops/15778/presubmit-kops-aws-scale-amazonvpc-using-cl2/1691438764041179136)).

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 1]
```diff
@@ -95,15 +95,19 @@ fi
...
+create_args+=("--set spec.kubeControllerManager.endpointSliceUpdatesBatchPeriod=500ms")
 create_args+=("--node-count=${KUBE_NODE_COUNT:-101}")
 # TODO: track failures of tests (HostPort & OIDC) when using `--dns=none`
 create_args+=("--dns none")
 create_args+=("--node-size=c6g.medium")
 create_args+=("--control-plane-count=${CONTROL_PLANE_COUNT:-1}")
 create_args+=("--master-size=${CONTROL_PLANE_SIZE:-c6g.2xlarge}")
-create_args+=("--zones=us-east-2a,us-east-2b,us-east-2c")
+create_args+=("--zones=us-east-2a")
+create_args+=("--set spec.networking.subnets[0].cidr=10.0.0.0/16")
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:110]:
```suggestion
```

**Comment by hakman** [comment]
@prateekgogia Can you squash the commits please? The failed scale test prevents the merge.

File: tests/e2e/scenarios/scalability/run-test.sh [hunk 2]
```diff
@@ -87,6 +87,10 @@ echo "ADMIN_ACCESS=${ADMIN_ACCESS}"
 #create_args="--networking cilium"
 create_args=()
 create_args=("--network-cidr=10.0.0.0/16")
+create_args=("--network-cidr=10.1.0.0/16")
+create_args=("--network-cidr=10.2.0.0/16")
+create_args=("--network-cidr=10.3.0.0/16")
+create_args=("--network-cidr=10.4.0.0/16")
```
Comment (hakman) [inline tests/e2e/scenarios/scalability/run-test.sh:93]:
```suggestion
create_args=("--network-cidr=10.0.0.0/16,10.1.0.0/16,10.2.0.0/16,10.3.0.0/16,10.4.0.0/16")
```

**Review by hakman — APPROVED** [review]

**Pull request merged**